### PR TITLE
NGLS-2999: Downgrade dependencies to lint Superset Python code

### PR DESCRIPTION
**Pull request by MichaelDinh**

### CODE REVIEW

File: requirements/testing.txt [hunk 1]
```diff
@@ -84,7 +84,7 @@ pyfakefs==4.5.6
     # via -r requirements/testing.in
 pyhive[presto]==0.6.5
     # via apache-superset
-pylint==2.17.3
+pylint==2.9.6
```
Comment (MichaelDinh) [inline requirements/testing.txt:85]: Seems like using a newer version of pylint runs up against a bunch of problems linting with the base superset repo.

File: requirements/base.txt [hunk 2]
```diff
@@ -313,7 +313,7 @@ werkzeug==2.3.2
     #   flask
     #   flask-jwt-extended
     #   flask-login
-wrapt==1.15.0
+wrapt==1.12.0
```
Comment (MichaelDinh) [inline requirements/base.txt]: Conflicting dependency, I didn't find wrapt used as a dependency anywhere so I'm not sure if the versioning matters?

Comment (rpastro) [reply requirements/base.txt]: You are not supposed to manually change the txt files. The dependencies in base.txt are controlled via setup.py.  You can see the instructions on how to regenerate the txt files at CONTRIBUTING.md.

In a nutshell:
```
$ python3 -m venv venv
$ source venv/bin/activate
$ python3 -m pip install -r requirements/integration.txt
$ pip-compile-multi --no-upgrade
```

File: requirements/testing.txt (hunk 1 above)
Comment (rpastro) [reply requirements/testing.txt:85]: You must control these dependencies in testing.in file.

**Review by rpastro — CHANGES_REQUESTED** [review]

File: requirements/testing.txt (hunk 1 above)
Comment (MichaelDinh) [reply requirements/testing.txt:85]: Not sure if this is followed by the superset guys: 


They manually bump the pylint dependency here:
https://github.com/apache/superset/pull/16146/commits/8c3d98656608429b204cbc5bdd8d80427744f6f9


and straight up got rid of the dependency two days ago:

https://github.com/apache/superset/commit/9fd936c18c854a5a3c2f14c5cf13bd49e0fda6ab


As of today, the github actions runner is still using 2.9.6

https://github.com/apache/superset/actions/runs/4941268247/jobs/8833730890


I'm genuinely confused if there's a structure to this or not.

Comment (rpastro) [reply requirements/testing.txt:85]: I've been using pip-compile-multi to regenerate the dependencies, so if we don't properly set the dependency in testing.in then the changes in testing.txt might be overwritten next time I update the packages.

Comment (MichaelDinh) [reply requirements/testing.txt:85]: Sure, I hardcoded in the versions that are pointed to in the main branch repo,

`pylint==2.9.6` 
`wrapt==1.12.1`

into the .in files for now until I can get a better understanding of what they're doing over there.

**Review by rpastro — APPROVED** [review]

**Pull request merged**